pkg/client/client.go

```diff
@@ -6,18 +6,25 @@ import (
 	"log/slog"
 
 	camelv1 "github.com/apache/camel-k/pkg/apis/camel/v1"
+	camelv1alpha "github.com/apache/camel-k/pkg/apis/camel/v1alpha1"
 	camelclient "github.com/apache/camel-k/pkg/client"
 	"github.com/go-logr/logr/slogr"
 	controllerlog "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 var logger = slog.Default().With(slog.String("component", "k8s-client"))
 
-type Client struct {
-	camelclient.Client
+type ScoClient interface {
```

Comment:
nitpick: in kubernetes a common pattern is to name this kind of things, `Interface`  and the related factory method `New` which would result in something like:

```go
var client sco.Interface 
client = sco.New()
```

I don't know if this is the pattern/naming/style we want to follow tough.
Food for thought